### ISSUE TITLE
Configuration ASP.NET Core Web Application (.Net Framework)

### ISSUE THREAD

**Issue by lacota** [MarimerLLC/cslaforum#224]
Sorry about my confusion but I am trying to setup my dataportal config info in a ASP.NET core web application, referencing the standard .net framework. Since it no longer uses the web.config file I have placed the settings in my appsettings.json;
{

    "CslaAuthentication": "Csla",
    "CslaDataPortalProxy": "Csla.DataPortalClient.WcfProxy, Csla",
    "CslaDataPortalUrl": "https://dataportal.hpass.corp.lacota.com/WcfPortal.svc"
}

But csla does not load them. I have tried placing them within an AppSettings node but Configuration.AppSettings still can't find them which is why, I am assuming, csla can't find them. 

How should we setup a reference to the dataportal in the new config paradigm?

Thanks

**Comment by rockfordlhotka** [comment]
Yes, this is an issue. I am still not sure about the best way to solve the issue either, because they have changed the app settings config technology quite a lot.

I have seen where people have open source AppSettings libraries on GitHub - basically to replace the original functionality. I'm not sure that I want to take a dependency on one of them. Nor am I sure I want to recreate their work, though that might be the right answer.

Adapting CSLA to the new model requires a lot of serious work that seems like it will be hard/impossible to do while keeping the same codebase as full .NET, though this still might be the right answer.

In the meantime, the "workaround" (which _also_ might be the long-term right answer) is for you to do your own config in the normal ASP.NET Core app startup code. All the CSLA settings can be set via code, so you can use the new .NET Core config technique to load the settings from your config file and then set the CSLA config properties using those values.

Even though this seems a little clunky, I think it might be the correct answer overall, because it gives you full flexibility to use the new (very powerful) config subsystem in ASP.NET Core in whatever way you see fit (and there are a _lot_ of options), and CSLA won't limit how you use it.

**Comment by rockfordlhotka** [comment]
The upcoming 4.7.200 release of CSLA includes `Csla.Configuration`, which abstracts `System.Configuration` on full .NET, and offers a nice bridge for use in .NET Core and other .NET Standard 2.0 environments.
